### ISSUE TITLE
Great Library

### ISSUE THREAD

**Issue by richteel** [carlosefr/magstripelib#3]
@carlosefr, thanks for this library. I did run into an issue with the card readers that I have but was able to fix it using a capacitor to filter the data line. I wrote up a blog post about it. I plan to make some changes to the library as I have an idea to make it more reliable for this type of issue.

The blog post is at https://teelsys.com/?p=708.

**Comment by MonsieurV** [comment]
Hi,

Concerning the issue you encountered, maybe using the input mode `INPUT_PULLUP` (for the [two pins](https://github.com/carlosefr/magstripelib/blob/master/MagStripe.cpp#L60) used with interrupt) will fix the issue (without requiring to add an external capacitor)?
See:
* https://www.arduino.cc/en/Tutorial/DigitalPins
* https://www.arduino.cc/en/Tutorial/DigitalInputPullup

PS. Great blog post @richteel ; and great library @carlosefr  :)

**Comment by carlosefr** [comment]
Thanks. 🙂

@richteel WRT having an interrupt attached to the data line (as you mention in your blog post), the reason for this is to avoid spending too much time inside the clock interrupt hander. Reading the data pin from there in a compatible way (i.e. using `digitalRead()` instead of the port registers) would be rather expensive.

**Comment by richteel** [comment]
Thanks @MonsieurV & @carlosefr, I added a new blog post with the changes I made to the code. In the blog I go over the changes I made and why I made the changes. I at the end, I state that anyone needing to read from a single card reader should use the magstripelib and why but if they need read from more than one reader they may want to look at the code I posted. @carlosefr , your work on this library made it very easy to make changes and modify the code for my needs. If I did not come across this, I may have trashed the card readers. Now they will sit on my completed projects shelf until there is a want to read a card. ;-)
